### PR TITLE
Added arc/circle drawing

### PR DESCRIPTION
**Pull request by kkoser**
This adds the ability to draw colored arcs and circles of varying radius
to the screen. This was tested by drawing circles of various sizes, one
of which I left in main as a proof of function. The functions use sin
and cos to determine the location of pixels to color as theta is
gradually ranged from start_t to end_t

Tested by drawing several circles on the screen.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -51,7 +51,10 @@ process_init() is a big step.  This initializes the process table, but also give
 
     //change text color to white after bootup
     console_set_fgcolor(255,255,255);
-
+
```
Comment (alexmwu) [inline src/main.c]: remove these lines from the PR

File: src/graphics.h [hunk 2]
```diff
@@ -18,6 +18,33 @@ struct graphics_color {
 int graphics_width();
 int graphics_height();
 
+/**
+ * @brief Draws a curve with given radius and range
+ * @details This draws a curve from start_t to end_t with
+ * a given radius. The curve will be circular as opposed to elliptical.
+ * Angles should be given on a unit circle ranging from 0 to 2PI
+ *
```
Comment (shuyangli) [inline src/graphics.h]: Please get rid of trailing space

File: src/graphics.h [hunk 3]
```diff
@@ -18,6 +18,33 @@ struct graphics_color {
 int graphics_width();
 int graphics_height();
 
+/**
+ * @brief Draws a curve with given radius and range
+ * @details This draws a curve from start_t to end_t with
+ * a given radius. The curve will be circular as opposed to elliptical.
+ * Angles should be given on a unit circle ranging from 0 to 2PI
+ * 
+ * @param x The x coordinate of the center of the arc
```
Comment (shuyangli) [inline src/graphics.h]: Please get rid of trailing space

File: src/graphics.h [hunk 4]
```diff
@@ -18,6 +18,33 @@ struct graphics_color {
...
+ * @param graphics_color The color to draw in
+ */
+void graphics_arc(int x, int y, double r, double start_theta, double end_theta,
+				  struct graphics_color c);
+
+/**
+ * @brief Draws a circle centered at given point
+ * @details Draws an arc from 0 to 2PI centered at x, y in the given color
+ * with a radius of r
+ *
```
Comment (shuyangli) [inline src/graphics.h]: Please get rid of trailing space

File: src/graphics.h (hunk 2 above)
Comment (kkoser) [inline src/graphics.h]: fixed

File: src/graphics.h (hunk 3 above)
Comment (kkoser) [inline src/graphics.h]: fixed.

File: src/graphics.h (hunk 4 above)
Comment (kkoser) [inline src/graphics.h]: fixed.

**Pull request merged**